src/Api/DataProvider/CoverCollectionDataProvider.php

```diff
@@ -36,12 +37,33 @@ public function getCollection(string $resourceClass, string $operationName = nul
         $isIdentifiers = $this->getIdentifiers($request);
 
         $query = $this->buildElasticQuery($identifierType, $isIdentifiers);
-        $searchResponse = $this->index->request('_search', \Elastica\Request::POST, $query->toArray());
-        $results = $searchResponse->getData();
+
+        // Note that we here don't uses the elastica request function to post the request to elasticsearch has we have
+        // had strange performance issues with it. We get information about the index and create the curl call by hand.
+        // We can do this as we now the complete setup and what should be taken into consideration.
+        $index = $this->index->getIndex();
+        $connection = $index->getClient()->getConnection();
+        $path = $index->getName().'/search/_search';
+        $url = $connection->hasConfig('url') ? $connection->getConfig('url') : '';
+        $json_query = JSON::stringify($query->toArray(), JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES);
+
+        $startQueryTime = microtime(true);
+        $ch = curl_init($url.$path);
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'POST');
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $json_query);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_HTTPHEADER, [
+            'Content-Type: application/json',
+            'Content-Length: '.strlen($json_query),
+        ]);
+        $response = curl_exec($ch);
```

Comment:
cUrl error handling?

Reply:
Not sure how to handle errors at this point.... should i logge the error?

Reply:
It will result in an error for the user, is the reported correctly?
Logging it would be good as well.